src/zc/relation/tokens.txt

```diff
@@ -332,13 +469,18 @@ this were a real system.  The organization will be a direct object reference.
     ...         return "<Roles instance (%s has %s in %s)>" % (
     ...             self.principal_id, ', '.join(self.role_ids),
     ...             self._organization.title)
-    ...     def __cmp__(self, other):
-    ...         return cmp(
-    ...             (self.principal_id, tuple(self.role_ids),
-    ...              self._organization.title),
-    ...             (other.principal_id, tuple(other.role_ids),
-    ...              other._organization.title))
-    ...
+    ...     def __lt__(self, other):
+    ...         _self = (
+    ...             self.principal_id,
+    ...             tuple(self.role_ids),
+    ...             self._organization.title,
+    ...         )
+    ...         _other = (
+    ...             other.principal_id,
+    ...             tuple(other.role_ids),
+    ...             other._organization.title,
+    ...         )
+    ...         return _self <_other
```

Comment:
I don't like seeing only one of six rich comparison methods being defined. If you use @functools.total_ordering, you can get away by defining two of six.